lib/Frontend/nvqpp/ConvertExpr.cpp

```diff
@@ -2234,218 +2252,218 @@ bool QuakeBridgeVisitor::TraverseCXXConstructExpr(clang::CXXConstructExpr *x,
-          backTy = ptrTy.getElementType();
-        if (backTy.isa<cc::CallableType>()) {
-          // Skip this constructor (for now).
-          return true;
-        }
-        if (auto stTy = dyn_cast<cc::StructType>(backTy)) {
-          if (!stTy.getMembers().empty()) {
-            // TODO: We don't support a callable class with data members yet.
-            TODO_loc(loc, "callable class with data members");
-          }
-          // Constructor generated as degenerate reference to call operator.
-          auto *fromTy = x->getArg(0)->getType().getTypePtr();
-          // FIXME: May need to peel off more than one layer of sugar?
-          if (auto *elabTy = dyn_cast<clang::ElaboratedType>(fromTy))
-            fromTy = elabTy->desugar().getTypePtr();
-          auto *fromDecl =
-              dyn_cast_or_null<clang::RecordType>(fromTy)->getDecl();
-          if (!fromDecl)
-            TODO_loc(loc, "recovering record type for a callable");
-          auto *objDecl = dyn_cast_or_null<clang::CXXRecordDecl>(fromDecl);
-          if (!objDecl)
-            TODO_loc(loc, "recovering C++ declaration for callable");
-          auto *callOperDecl = findCallOperator(objDecl);
-          if (!callOperDecl) {
-            auto &de = mangler->getASTContext().getDiagnostics();
-            auto id = de.getCustomDiagID(
-                clang::DiagnosticsEngine::Error,
-                "std::function initializer must be a callable");
-            de.Report(x->getBeginLoc(), id);
-            return true;
-          }
-          auto kernelCallTy = cast<cc::CallableType>(ctorTy);
-          auto kernelName = generateCudaqKernelName(callOperDecl);
-          popValue(); // replace value at TOS.
-          return pushValue(builder.create<cc::CreateLambdaOp>(
-              loc, kernelCallTy, [&](OpBuilder &builder, Location loc) {
-                auto args = builder.getBlock()->getArguments();
-                auto call = builder.create<func::CallOp>(
-                    loc, kernelCallTy.getSignature().getResults(), kernelName,
-                    args);
-                builder.create<cc::ReturnOp>(loc, call.getResults());
-              }));
+      }
+    }
+  } else if (isInNamespace(ctor, "std")) {
+    auto isVectorOfQubitRefs = [&]() {
+      if (auto *ctor = x->getConstructor()) {
+        if (isInNamespace(ctor, "std") && ctor->getNameAsString() == "vector") {
+          if (valueStack.empty())
+            return false;
+          Value v = peekValue();
+          return v && isa<quake::VeqType>(v.getType());
         }
       }
-      if (ctorName == "reference_wrapper") {
-        // The `reference_wrapper` class is used to guide the `qudit&` through a
-        // container class (like `std::vector`). It is a NOP at the Quake level.
-        [[maybe_unused]] auto tosTy = peekValue().getType();
-        assert((isa<quake::RefType, quake::VeqType>(tosTy)));
+      return false;
+    };
```

Comment:
nit - I guess you could just evaluate this here and treat `isVectorOfQubitRefs` as a `bool` instead of a lambda to be called.

Reply:
[366cb3f](https://github.com/NVIDIA/cuda-quantum/pull/1068/commits/366cb3fd5ad550d3fb37f614041fe7bc4bc58a73)